### ISSUE TITLE
Using CoroutineName as coroutine context leads to "java.lang.IllegalArgumentException: Must be used with DispatchedContinuation" on coroutine resume 

### ISSUE THREAD

**Issue by wild-lynx** [Kotlin/kotlinx.coroutines#133]
Run the following code:
```kotlin
fun main(args: Array<String>) = runBlocking(CoroutineName("channels.main")) {
    val c = async(CoroutineName("c coroutine")) {
        delay(500) // exception will be thrown while resuming
        42
    }
    val res = c.await()
}
```

You will get an exception:
> Exception in thread "kotlinx.coroutines.DefaultExecutor" java.lang.IllegalArgumentException: Must be used with DispatchedContinuation
	at kotlinx.coroutines.experimental.CancellableContinuationImpl.resumeUndispatched(CancellableContinuation.kt:252)
	at kotlinx.coroutines.experimental.EventLoopBase$DelayedResumeTask.run(EventLoop.kt:237)
	at kotlinx.coroutines.experimental.EventLoopBase.processNextEvent(EventLoop.kt:129)
	at kotlinx.coroutines.experimental.DefaultExecutor.run(DefaultExecutor.kt:54)
	at java.lang.Thread.run(Thread.java:748)

The example is taken and reduced from [here](https://github.com/Kotlin/kotlinx.coroutines/blob/master/coroutines-guide.md#naming-coroutines-for-debugging)

1.1.50-master-dev + IDEA 2017.2.5 RC IU-172.4343.5
1.2-Beta-eap-21 + IDEA 2017.2.5 RC IU-172.4343.5 | IDEA 2017.3 EAP IC-173.2463.16

**Comment by elizarov** [comment]
Fixed in develop branch

**Comment by elizarov** [comment]
Released in version 0.19
